Sources/BuildSystemIntegration/ExternalBuildSystemAdapter.swift

```diff
@@ -226,19 +225,17 @@ actor ExternalBuildSystemAdapter {
         try? FileManager.default.contentsOfDirectory(at: buildServerConfigLocation, includingPropertiesForKeys: nil)
         .filter { $0.pathExtension == "json" }
 
-      if let configFileURL = jsonFiles?.sorted(by: { $0.lastPathComponent < $1.lastPathComponent }).first,
-        let configFilePath = AbsolutePath(validatingOrNil: try? configFileURL.filePath)
-      {
-        return configFilePath
+      if let configFileURL = jsonFiles?.sorted(by: { $0.lastPathComponent < $1.lastPathComponent }).first {
+        return configFileURL
       }
     }
 
     // Pre Swift 6.1 SourceKit-LSP looked for `buildServer.json` in the project root. Maintain this search location for
     // compatibility even though it's not a standard BSP search location.
     if let workspaceFolder = workspaceFolder,
-      localFileSystem.isFile(workspaceFolder.appending(component: "buildServer.json"))
+      FileManager.default.fileExists(at: workspaceFolder.appendingPathComponent("buildServer.json"))
     {
-      return workspaceFolder.appending(component: "buildServer.json")
+      return workspaceFolder.appendingPathComponent("buildServer.json")
```

Comment:
Perhaps create a local variable for this and re-use it?

Reply:
Good suggestion 👍🏽